### PR TITLE
Add `dispatcher` to HandleHolder

### PR DESCRIPTION
**Pull request by jasonwyatt**
Provides a particle-wide dispatcher which can be used to access any/all of a particle's handles safely on the Scheduler when not in a lifecycle method or handle callback.

```kotlin
withContext(myParticle.handles.dispatcher) {
    // Do whatever with myParticle's handles
}
```

### CODE REVIEW

**Comment by jblebrun** [comment]
Handy!

File: javatests/arcs/android/entity/DifferentHandleManagerTest.kt [hunk 1]
```diff
@@ -90,6 +90,12 @@ class DifferentHandleManagerTest : HandleManagerTestBase() {
         super.singleton_dereferenceEntity()
     }
 
+    @Ignore("b/154947352 - Deflake")
+    @Test
+    override fun collection_dereferenceEntity_nestedReference() {
+        super.collection_dereferenceEntity_nestedReference()
+    }
+
```
Comment (cromwellian) [inline javatests/arcs/android/entity/DifferentHandleManagerTest.kt:98]: Was this made flaky as the result of this PR, or was it always flaky?

Comment (jasonwyatt) [reply javatests/arcs/android/entity/DifferentHandleManagerTest.kt:98]: It's been flaky for a while, I've just been marking them as such when I come across them

**Pull request merged**